### PR TITLE
National Library of Poland ISBN search

### PR DESCRIPTION
**Pull request by Eccenux**
Hi. I've added BN search for Zotero, to be used hopefully on Wikipedia (Citoid) in future.

This is a simple marcXml search and BN API is rather stable. I've created the test in the Scaffold IDE and it seems to parse data without problems.

### CODE REVIEW

**Comment by Eccenux** [comment]
I just noticed I've copy&pasted wrong `browserSupport`. Not sure if that matters. Using ES6, so no IE support.

File: National Library of Poland ISBN.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+{
+	"translatorID": "aa7f310e-10d3-4209-91dc-88301e7070c6",
+	"label": "National Library of Poland ISBN",
+	"creator": "Maciej Nux Jaros",
+	"target": "",
+	"minVersion": "4.0",
+	"maxVersion": "",
+	"priority": 97,
```
Comment (adam3smith) [inline National Library of Poland ISBN.js:8]: We'd want the priority lower -- I think this should probably run after the K10+ catalog or after the BnF

**Comment by adam3smith** [comment]
Thanks -- looks good in general. 
There is a larger question about how many different ISBN searches we're going to add. I think in general this one makes sense -- we don't have good coverage of Poland/Eastern Europe in the current ISBN searches -- but this is run on every single ISBN that people input and makes things a bit slower, so there is a real downside to keep adding these.
@dstillman what's your view on this? One option would be to just run this on ISBNs with Polish [region code](https://en.wikipedia.org/wiki/List_of_ISBN_registration_groups) -- that's surely going to miss some stuff (e.g. other Eastern countries well represented in the NLP) but it would avoid the issue above.

File: National Library of Poland ISBN.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	"inRepository": true,
+	"translatorType": 8,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-05-20 15:41:08"
+};
```
Comment (adam3smith) [inline National Library of Poland ISBN.js:13]: that's the ; the linter is complaining about

**Review by adam3smith — CHANGES_REQUESTED** [review]

File: National Library of Poland ISBN.js (hunk 1 above)
Comment (Eccenux) [reply National Library of Poland ISBN.js]: Lower priority is higher number? Changed to the same as bnf (98)

File: National Library of Poland ISBN.js (hunk 2 above)
Comment (Eccenux) [reply National Library of Poland ISBN.js]: Should be fine now. That came from editing in VSC.

**Comment by Eccenux** [comment]
> ...One option would be to just run this on ISBNs with Polish [region code](https://en.wikipedia.org/wiki/List_of_ISBN_registration_groups) -- that's surely going to miss some stuff (e.g. other Eastern countries well represented in the NLP) but it would avoid the issue above.

I happen to have access to full dump of BN. I'll check for patterns in ISBN.

**Comment by Eccenux** [comment]
Filtering by `/^(97883|83)/` might work. Stats from the database:
* pl83: 1 085 412
* total with ~valid isbn: 1 433 909

Stats SQL:
```
select 'pl83' as type, sum(isbn_count) from temp_isbn_short
--select * from temp_isbn_short
where isbn_prefix ~ '^97883' or isbn_prefix ~ '^83'
union all
select 'total with ~valid isbn', sum(isbn_count) from temp_isbn_short
;
```

Prefix data of BN:
[data-isbn-short.csv](https://github.com/zotero/translators/files/11523245/data-isbn-short.csv)

**Comment by adam3smith** [comment]
Oh yes, I think that's a strong case for prefiltering ISBNs. (which would then also allow us to add many more ISBN translators). @dstillman would it make more sense to do this in the identifier code or the translators individually (which I think we can already do?)

**Comment by dstillman** [comment]
Much better to do it in the translator detect code — then it'll just work everywhere.

**Comment by Eccenux** [comment]
Done. Added a simple filter which should be fast.

**Comment by mvolz** [comment]
> Thanks -- looks good in general. There is a larger question about how many different ISBN searches we're going to add. I think in general this one makes sense -- we don't have good coverage of Poland/Eastern Europe in the current ISBN searches -- but this is run on every single ISBN that people input and makes things a bit slower, so there is a real downside to keep adding these. @dstillman what's your view on this? One option would be to just run this on ISBNs with Polish [region code](https://en.wikipedia.org/wiki/List_of_ISBN_registration_groups) -- that's surely going to miss some stuff (e.g. other Eastern countries well represented in the NLP) but it would avoid the issue above.

Sorry for my ignorant question, but there is any way to do this less syncronously to avoid this issue? Or is that too hard with the current infrastructure. 

There is a long standing request for us to support the national library in Japan which I was hoping to look into next. https://phabricator.wikimedia.org/T192528

**Comment by Eccenux** [comment]
@adam3smith Hi. I think I resolved everything here, but please let me know if there is more work for me to do? Otherwise, could you re-run tests and merge?

**Comment by Eccenux** [comment]
Hi, I've updated the base for my pull request so merging should be easy.

Maybe this linter will work too... Although I don't know where the version number '65161a4967b2...' came from in the script. From what I see it's some change from `2022-10-28`:
OUP: Lint, add authors from DOM if necessary (#2898)

**Comment by adam3smith** [comment]
(I believe CI fails if there have been any CI-relevant changes on the `master` branch while the PR is under review)

**Comment by adam3smith** [comment]
Sorry for the delay -- I wanted to spend some time testing this to make sure this works as intended, since we haven't used the idea of restricting to ISBN regions before -- it's looking great and I'm very excited that this means we'll be able to add other national (or quasi-national) libraries to ISBN search without real costs.

**Pull request merged**